python/ray/experimental/channel/local_channel.py

```diff
@@ -0,0 +1,54 @@
+import uuid
+from typing import Any
+
+import ray
+from ray.experimental.channel import ChannelContext
+from ray.experimental.channel.common import ChannelInterface
+from ray.util.annotations import PublicAPI
+
+
+@PublicAPI(stability="alpha")
+class LocalChannel(ChannelInterface):
```

Comment:
nit: Prefer a name like NopChannel to emphasize that the channel does not do anything. LocalChannel kind of sounds like it should work as long as two tasks are on the same local node, though not necessarily in the same address space.

Reply:
Or `ProcessLocalChannel`?

Reply:
`IntraProcessChannel` sounds good.

Reply:
Rename `LocalChannel` to `IntraProcessChannel` https://github.com/ray-project/ray/pull/45591/commits/d1a025c050cb80500d76da8f53ff49f0fc2fe0f1. 

@stephanie-wang do we need to:

* (1) Rename `MultiChannel._local_channel` to `MultiChannel._intra_process_channel`. I am not sure whether `IntraProcessChannel` is the only "local channel", or if it is possible to have different kinds of "local channels".
* (2) Rename `local_channel.py` to `intra_process_channel.py`.

Reply:
Updated https://github.com/ray-project/ray/pull/45591/commits/af458bff73a2ef8a4972fa013e280b2662fcc5d9